### PR TITLE
Update agent header

### PR DESCRIPTION
**Pull request by michaelbausor**
Toolkit PR: https://github.com/googleapis/toolkit/pull/1042

### CODE REVIEW

**Comment by michaelbausor** [comment]
Updated with appName->libName change, PTAL

File: src/AgentHeaderDescriptor.php [hunk 1]
```diff
@@ -43,44 +41,87 @@ class AgentHeaderDescriptor
...
-     *     @type string $clientVersion the version of the client application.
+     *     @type string $phpVersion the PHP version.
+     *     @type string $libName the name of the client application.
+     *     @type string $libVersion the version of the client application.
      *     @type string $codeGenName the code generator name of the client library.
      *     @type string $codeGenVersion the code generator version of the client library.
      *     @type string $gaxVersion the GAX version.
-     *     @type string $phpVersion the PHP version.
+     *     @type string $grpcVersion the gRPC version.
+     *     @type string $additionalMetrics a dictionary of additional metrics for the header.
```
Comment (bjwatson) [inline src/AgentHeaderDescriptor.php]: Scratch `$additionalMetrics`. The service control team wants tight control over what we pass. The only thing we should leave somewhat open is `$libName`.

File: src/AgentHeaderDescriptor.php [hunk 2]
```diff
@@ -43,44 +41,87 @@ class AgentHeaderDescriptor
...
+        $metricsHeaders['gl-php'] = $phpVersion;
+
+        if (isset($headerInfo['libName'])) {
+            $clientVersion = isset($headerInfo['libVersion'])
+                ? $headerInfo['libVersion']
+                : AgentHeaderDescriptor::UNKNOWN_HEADER;
+            $metricsHeaders[$headerInfo['libName']] = $clientVersion;
+        }
+
+        if (isset($headerInfo['codeGenName'])) {
```
Comment (bjwatson) [inline src/AgentHeaderDescriptor.php]: This should always be set, and should always be `gapic`. We're not worried about the use case of anyone else using GAX directly.

File: tests/AgentHeaderDescriptorTest.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        $agentHeaderDescriptor = new AgentHeaderDescriptor([]);
+        $header = $agentHeaderDescriptor->getHeader();
+
+        $this->assertSame($expectedHeader, $header);
+    }
+
+    public function testWithInput()
+    {
+        $expectedHeader = [AgentHeaderDescriptor::AGENT_HEADER_KEY => [
+            'gl-php/php-9.9.9 gccl/gccl-9.9.9 gapic/gapic-9.9.9 gax/gax-9.9.9 grpc/grpc-9.9.9' .
```
Comment (bjwatson) [inline tests/AgentHeaderDescriptorTest.php]: These versions are misleading. They should just be `9.9.9`. Our spec for the custom header field says these _should_ be semver (i.e. they don't have to be, but they ought to be if possible).

File: src/AgentHeaderDescriptor.php (hunk 1 above)
Comment (michaelbausor) [inline src/AgentHeaderDescriptor.php]: Done

File: src/AgentHeaderDescriptor.php (hunk 2 above)
Comment (michaelbausor) [inline src/AgentHeaderDescriptor.php]: Done. Removed codeGenName and renamed codeGenVersion to gapicVersion

File: tests/AgentHeaderDescriptorTest.php (hunk 3 above)
Comment (michaelbausor) [inline tests/AgentHeaderDescriptorTest.php]: Done. Updated to different numbers to test assignment.

File: src/AgentHeaderDescriptor.php [hunk 4]
```diff
@@ -52,11 +52,9 @@ class AgentHeaderDescriptor
      *     @type string $phpVersion the PHP version.
      *     @type string $libName the name of the client application.
      *     @type string $libVersion the version of the client application.
-     *     @type string $codeGenName the code generator name of the client library.
-     *     @type string $codeGenVersion the code generator version of the client library.
+     *     @type string $gapicVersion the code generator version of the client library.
```
Comment (bjwatson) [inline src/AgentHeaderDescriptor.php]: Nit: I'd specify that this is "...the code generator version of the _GAPIC_ library."

File: src/AgentHeaderDescriptor.php [hunk 5]
```diff
@@ -43,44 +41,74 @@ class AgentHeaderDescriptor
     // TODO(michaelbausor): include bumping this version in a streamlined
     // release process. Issue: https://github.com/googleapis/gax-php/issues/48
     const GAX_VERSION = '0.7.0';
+    const UNKNOWN_HEADER = '';
```
Comment (bjwatson) [inline src/AgentHeaderDescriptor.php]: Let's rename this to `UNKNOWN_VERSION`. Sorry I didn't catch this in my first review.

Comment (michaelbausor) [inline src/AgentHeaderDescriptor.php]: Done

File: src/AgentHeaderDescriptor.php (hunk 4 above)
Comment (michaelbausor) [inline src/AgentHeaderDescriptor.php]: Done

**Pull request merged**